code/parachain/frame/cosmwasm/src/mock.rs

```diff
@@ -197,29 +197,36 @@ impl pallet_timestamp::Config for Test {
 
 /// Native <-> Cosmwasm account mapping
 pub struct AccountToAddr;
+
 impl Convert<alloc::string::String, Result<AccountId, ()>> for AccountToAddr {
 	fn convert(a: alloc::string::String) -> Result<AccountId, ()> {
-		match a.strip_prefix("0x") {
-			Some(account_id) => Ok(<[u8; 32]>::try_from(hex::decode(account_id).map_err(|_| ())?)
-				.map_err(|_| ())?
-				.into()),
-			_ => Err(()),
-		}
+		let account =
+			ibc_primitives::runtime_interface::ss58_to_account_id_32(&a).map_err(|_| ())?;
+		Ok(account.into())
 	}
 }
+
 impl Convert<AccountId, alloc::string::String> for AccountToAddr {
 	fn convert(a: AccountId) -> alloc::string::String {
-		alloc::format!("0x{}", hex::encode(a))
+		let account = ibc_primitives::runtime_interface::account_id_to_ss58(a.into(), 49);
```

Comment:
https://github.com/ComposableFi/centauri/blob/ca553f0d1c6c35b00ebf4fb4f5c43fa33b09d895/contracts/pallet-ibc/primitives/src/runtime_interface.rs#L77

Reply:
it is a private function, the public api export `Vec<u8>` so it has to be converted to string